src/app/core/auth/state/auth.selectors.ts

```diff
@@ -0,0 +1,20 @@
+import { Selector } from '@ngxs/store'
+import { Tokens } from 'src/app/core/auth/types/tokens'
+import { AuthStateModel, AuthStateService } from 'src/app/core/auth/state/auth-state.service'
+
+export class AuthStateSelectors {
+  @Selector([AuthStateService])
+  public static token({ tokens }: AuthStateModel): Tokens {
+    return tokens
+  }
+
+  @Selector([AuthStateService])
+  public static isAuthenticated({ isAuthenticated }: AuthStateModel): boolean {
+    return isAuthenticated
+  }
+
+  @Selector([AuthStateService])
+  public static isLoginFormSubmitted({ isLoginFormSubmitted }: AuthStateModel): boolean {
+    return isLoginFormSubmitted
+  }
```

Comment:
Whatever you do, you don't need to store a flag if forrm has been submitted. Remove it from the state and do not use it anywhere else. Try to implement your feature differently.